### PR TITLE
Implement Security Area Management

### PR DESCRIPTION
**Pull request by prescornic**
# Implement Security Area Management

## Description
This PR introduces comprehensive Security Area (Group) management to the `hikconnect_lib`. It implements full CRUD operations and arm/disarm capabilities for NVR areas by interacting directly with the Hik-Connect `v3` API endpoints. 

This ensures that Home Assistant integrations can rely on the library to handle area state modifications securely and robustly without polluting the integration with direct API calls.

## Why
This functionality is essential for Home Assistant users who want to create automations based on arming and disarming their security areas (e.g., auto-arming the cameras when leaving home). 

Required for [tomasbedrich/home-assistant-hikconnect#47](https://github.com/tomasbedrich/home-assistant-hikconnect/issues/47).

### Area CRUD Operations
* **`get_areas()` / `get_area()`**: Fetch all areas for a device and list their members.
* **`create_area()`**: Creates a new area with given resource/camera IDs.
* **`delete_area()`**: Uses the standard `DELETE /v3/devices/group/{device_serial}/{group_id}` endpoint. *(Note: Discovered that the `POST /v3/open/trust/v1/group/destroy` endpoint can fail with error code `70002` on offline devices, whereas the `DELETE` verb works consistently).*
* **`update_area()`**: Hik-Connect API returns `405 Method Not Allowed` for `PUT/PATCH` requests on groups. This function seamlessly handles updates by performing a **Delete → Recreate** cycle under the hood.

### Member Management & Automation
* **`edit_area_members()`**: A high-level helper to easily add or remove specific camera IDs from an existing area. 
  * Automatically calculates the delta and handles the delete+recreate requirement.
  * Ensures that if the last member is removed from an area, the area is automatically deleted (preventing invalid empty groups).

### Area Control (Defence Mode)
* **`set_area_defence_mode()`**: Base method to alter security area modes.
* **`arm_area()`**, **`arm_area_silent()`**, **`disarm_area()`**: Helpful abstractions to set standard modes (`1` = armed, `2` = armed silent, `0` = disarmed).

## Testing
* Added 30 comprehensive unit tests in `test_api.py` covering all new methods, mock HTTP sequences, and edge cases (e.g., verifying `delete_area` triggers properly when removing the last member). All tests pass.
* Verified live on NVR hardware.

## Notes for Reviewers
Because updating an area performs a recreation, the API assigns a **new** `group_id` upon update. The library’s `update_area` and `edit_area_members` functions are designed to return the new `group_id` in their response dictionaries so callers can update their local state.

### CODE REVIEW

**Comment by prescornic** [comment]
@tomasbedrich could you please approve build action https://github.com/tomasbedrich/hikconnect/actions/runs/29257193773 ? All tests were  fixed.

**Comment by prescornic** [comment]
Test fixed. Hope will pass build pipeline :)

**Comment by prescornic** [comment]
@tomasbedrich Could you please approve pipeline?

**Comment by prescornic** [comment]
@tomasbedrich Can we merge this ? So I can prepare integration part.

File: README.md [hunk 1]
```diff
@@ -54,6 +54,52 @@ async with HikConnect() as api:
...
+    #   {'group_id': 110548, 'device_serial': 'ZZZZZZZZZ', 'member_id': 'abc123...'},
+    #   ...
+    # ]
+    # member_id corresponds to camera id returned by get_cameras()
+
+    # Create a new area with specific cameras
+    cameras = [camera async for camera in api.get_cameras(my_device_serial)]
+    camera_ids = [c["id"] for c in cameras[:2]]  # first two cameras
+    result = await api.create_area(my_device_serial, "Front Gate", camera_ids)
+    new_group_id = result.get("groupId")  # or parse from result as needed
```
Comment (tomasbedrich) [inline README.md]: `create_area()` returns the normalized keys (`group_id`), not the raw API key, so `result.get("groupId")` always returns `None` here:

```suggestion
    new_group_id = result["group_id"]
```

File: README.md [hunk 2]
```diff
@@ -54,6 +54,52 @@ async with HikConnect() as api:
...
+    # Update an existing area (rename and/or change cameras)
+    await api.update_area(my_device_serial, my_group_id, "New Name", camera_ids)
+
+    # Arm / disarm an area
+    await api.arm_area(my_device_serial, my_group_id)           # mode=1
+    await api.arm_area_silent(my_device_serial, my_group_id)    # mode=2 (no beep)
+    await api.disarm_area(my_device_serial, my_group_id)        # mode=0
+
+    # Delete an area
+    await api.delete_area(my_group_id)
```
Comment (tomasbedrich) [inline README.md]: `delete_area()` takes `(device_serial, group_id)`, so this example raises `TypeError` when copy-pasted:

```suggestion
    await api.delete_area(my_device_serial, my_group_id)
```

File: hikconnect/api.py [hunk 3]
```diff
@@ -240,6 +242,305 @@ async def get_cameras(self, device_serial: str):
...
+        Use the convenience wrappers ``arm_area()``, ``arm_area_silent()``,
+        and ``disarm_area()`` instead of calling this directly.
+        """
+        payload = {"groupId": group_id, "mode": mode}
+        async with self.client.post(
+            f"{self.BASE_URL}/v3/devices/group/{device_serial}/switchDefenceMode",
+            json=payload,
+        ) as res:
+            res_json = await res.json()
+        log.debug("Got set defence mode response '%s'", res_json)
```
Comment (tomasbedrich) [inline hikconnect/api.py:504]: Unlike `delete_area()`, this doesn't check `meta.code`. The API tends to return HTTP 200 with an error code in `meta` (you hit `70002` yourself on offline devices), so arming an offline device would silently "succeed" — which matters for the auto-arm-when-leaving use case in HA.

**Review by tomasbedrich — APPROVED** [review]
Thanks, nice work! :) The code looks good and tests are solid. Approving with two small fixes for the README example (see suggestions) and one note about meta.code checking, which can be a follow-up.

File: README.md (hunk 1 above)
Comment (prescornic) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (prescornic) [reply README.md]: fixed

File: hikconnect/api.py (hunk 3 above)
Comment (prescornic) [reply hikconnect/api.py:504]: Thanks for notice. I missed this scenario in my tests. Now it detect offline device and rise an exception

**Comment by tomasbedrich** [comment]
LGTM, thank you. Released as 2.1.0
